src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1089,11 +1089,13 @@ private void OnFileChanged(object sender, FileSystemEventArgs e)
             // - the host.json file was changed
             // - a function.json file was changed
             // - a function directory was added/removed
+            // - a function directory was renamed
             string fileName = Path.GetFileName(e.Name);
             if (isWatchedDirectory ||
                 ((string.Compare(fileName, ScriptConstants.HostMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0) ||
                 string.Compare(fileName, ScriptConstants.FunctionMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0) ||
-                (Directory.EnumerateDirectories(ScriptConfig.RootScriptPath).Count() != _directoryCountSnapshot))
+                (Directory.EnumerateDirectories(ScriptConfig.RootScriptPath).Count() != _directoryCountSnapshot) ||
+                (e.ChangeType == WatcherChangeTypes.Renamed && String.IsNullOrEmpty(Path.GetExtension(e.FullPath))))
```

Comment:
Yep, that's a good approach.  One question with saving the snapshot as an IEnumerable, is it possible that lazy evaluation could provide incorrect results (ie snapshot isn't evaluated at snapshot time)?